src/App.tsx

```diff
@@ -15,36 +16,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SORT {
+  ALPHABET = 'alphabet',
+  LENGTH = 'length',
+  DEFAULT = '',
+}
+
+function sort(goods: string[], sortField: SORT, isReversed: boolean): string[] {
+  const sortedGoods = [...goods];
+
+  switch (sortField) {
+    case SORT.ALPHABET:
+      sortedGoods.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+
+    case SORT.LENGTH:
+      sortedGoods.sort((good1, good2) => good1.length - good2.length);
+      break;
+
+    case SORT.DEFAULT:
+    default:
+      break;
+  }
+
+  return isReversed ? sortedGoods.reverse() : sortedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SORT>(SORT.DEFAULT);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  const newGoodsFromServer = sort(goodsFromServer, sortField, isReversed);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': SORT.ALPHABET !== sortField,
+          })}
+          onClick={() => {
```

Comment:
The active button should be marked to indicate the current sort field to the user. It seems you have already implemented this by using conditional classes, which is great.